Source/Scene/PolylineCollection.js

```diff
@@ -1049,19 +1049,29 @@ import SceneMode from './SceneMode.js';
     function removePolylines(collection) {
         if (collection._polylinesRemoved) {
             collection._polylinesRemoved = false;
+            var definedPolylines = [];
+            var polyIndex=0;
+            var polyline;
+            var i;
 
-            var polylines = [];
-
-            var length = collection._polylines.length;
-            for ( var i = 0, j = 0; i < length; ++i) {
-                var polyline = collection._polylines[i];
+            for (i=0; i < collection._polylines.length; ++i) {
+                polyline = collection._polylines[i];
                 if (defined(polyline)) {
-                    polyline._index = j++;
-                    polylines.push(polyline);
+                    polyline._index = polyIndex++;
+                    definedPolylines.push(polyline);
+                }
+            }
+
+            var definedPolylinesToUpdate = [];
+            for (i=0; i < collection._polylinesToUpdate.length; ++i) {
+                polyline = collection._polylines[i];
+                if (defined(polyline) && defined(polyline._polylineCollection)) {
```

Comment:
I like that `polylinesToUpdate` filtering has moved here instead, but I wonder if it can be taken further.

This line in `remove` still kinda bugs me:
 `this._polylines[polyline._index] = undefined; // Removed later`.

Would it be cleaner to delete that line and filter `polylines` the same why `polylinesToUpdate` is filtered here, i.e. by checking that `polyline._polylineCollection` is defined? Or even better, add a private getter `isDestroyed` to `Polyline` that returns true if `polyline._polylineCollection` is undefined and use the getter instead.

Anywhere that checks if a polyline is defined would instead check `polyline.isDestroyed()`. `removePolylines`, `releaseShaders` and `destroyPolylines` would need to be updated.